consumer/pdata/log.go

```diff
@@ -14,14 +14,73 @@
 
 package pdata
 
-import otlplogs "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/logs/v1"
-
-// NewResourceLogsSliceFromOrig creates ResourceLogsSlice from otlplogs.ResourceLogs.
-// This function simply makes generated newResourceLogsSlice() function publicly
-// available for internal.data.Log to call. We intentionally placed data.Log in the
-// internal package so that it is not available publicly while it is experimental.
-// Once the experiment is over data.Log should move to this package (pdata) and
-// NewResourceLogsSliceFromOrig function will no longer be needed.
-func NewResourceLogsSliceFromOrig(orig *[]*otlplogs.ResourceLogs) ResourceLogsSlice {
-	return ResourceLogsSlice{orig}
+import (
+	"github.com/gogo/protobuf/proto"
+
+	otlplogs "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/logs/v1"
+)
+
+// This file defines in-memory data structures to represent logs.
+
+// Logs is the top-level struct that is propagated through the logs pipeline.
+//
+// This is a reference type (like builtin map).
+//
+// Must use NewLogs functions to create new instances.
+// Important: zero-initialized instance is not valid for use.
+type Logs struct {
+	orig *[]*otlplogs.ResourceLogs
+}
+
+// LogsFromProto creates the internal Logs representation from the ProtoBuf.
+func LogsFromProto(orig []*otlplogs.ResourceLogs) Logs {
```

Comment:
For naming consistency can we rename this to `LogsFromOtlp`? (We use `TracesFromOtlp` for traces).

Reply:
Done.